subscriber.rb

```diff
@@ -1,21 +1,47 @@
 require 'aws-sdk-sqs'
 require_relative 'load_aws'
 require_relative 'aws_sqs_client'
-
+require_relative 'q_it_errors'
 class Subscriber
 
   attr_reader :sqs, :sleep_period, :queue_url
 
   def initialize(options)
     @sqs = AwsSQSClient.new.client
-    @queue_url = options[0].chomp
-    @sleep_period = valid_sleep_period?(options[1]) ? options[1].to_i : 5
+    @queue_url = valid_sqs_url(options[0])
+    @sleep_period = validate_sleep_period(options[1])
```

Comment:
You could set these instance variables and then call the validation methods.
I'd suggest to use encapsulation for the individual validation methods and have a public interface to the main validation method.

Reply:
same here.. will be changing the implementation. Will help get rid of a lot of unnecessary code in the initialize methods for both publisher and subscriber.